src/components/BulkOperations/index.jsx

```diff
@@ -0,0 +1,208 @@
+import { useEffect, useState } from 'react';
+import {Link} from 'react-router-dom'
+import classNames from 'classnames';
+import Papa from 'papaparse';
+import {
+  SelectMenu, MenuItem, Dropzone, DataTable, Alert,
+} from '@openedx/paragon';
+
+import Collapsible from '../Collapsible';
+import LoadingSpinner from '../LoadingSpinner';
+import { formatDate } from '../../utils';
+import DiscoveryDataApiService from '../../data/services/DiscoveryDataApiService';
+import './BulkOperations.scss';
+
+const availableOperations = {
+  course_create: 'Bulk Create',
+  course_partial_update: 'Bulk Course Update',
+  course_run_partial_update: 'Bulk CourseRun Update',
+  course_rerun: 'Bulk Rerun',
+};
+
+const BulkOperations = () => {
+  const [bulkOperationId, setBulkOperationId] = useState(null);
+  const [fileContent, setFileContent] = useState(null);
+  const [file, setFile] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [submitSuccess, setSubmitSuccess] = useState(null);
+  const [historicalRecords, setHistoricalRecords] = useState(null);
+
+  function filteredHistoricalRecords() {
+    if (historicalRecords == null) {
+      return [];
+    }
+    return historicalRecords.filter(rec => !bulkOperationId || rec.task_type === bulkOperationId);
+  }
+
+  function handleUploadNew() {
+    setFileContent(null);
+    setFile(null);
+  }
+
+  function handleFileUpload({ fileData }) {
+    const file = fileData.get('file');
+    const reader = new FileReader();
+    reader.onload = (e) => {
+      const text = e.target.result;
+      setFile(file);
+      setFileContent(text);
+    };
+    reader.readAsText(file);
+  }
+
+  async function handleSubmit() {
+    setIsLoading(true);
+    try {
+      const res = await DiscoveryDataApiService.createBulkOperation(file, bulkOperationId);
+      setSubmitSuccess(true);
+      setHistoricalRecords([res.data, ...historicalRecords]);
+    } catch (e) {
+      setSubmitSuccess(false);
+    }
+    setIsLoading(false);
+  }
+
+  async function fetchBulkOpTasks() {
+    const response = await DiscoveryDataApiService.fetchBulkOperations();
+    setHistoricalRecords(response.data.results);
+  }
+
+  function getDropZone() {
+    return (
+      <Dropzone
+        onProcessUpload={handleFileUpload}
+      />
+    );
+  }
+
+  function getCollapsibleTitle() {
+    return <span className="font-weight-bold">Processing History</span>;
+  }
+
+  function getAlert() {
+    let variant; let
+      message;
+    if (submitSuccess == null) {
+      return null;
+    }
+
+    if (submitSuccess == true) {
+      variant = 'success';
+      message = 'Successfully submitted task for bulk operation';
+    } else if (submitSuccess == false) {
+      variant = 'danger';
+      message = 'Failed to submit task for processing';
+    }
```

Comment:
nit: Can we store these strings as constants and use them here instead? Hardcoding strings in the code doesn't look clean and makes maintenance harder.

Reply:
We only really need them here. And the `message` might change depending on the page/context, so I don't think they will provide much "reusability" if we factor them out somewhere.